### ISSUE TITLE
CoCSharp.Server - Implement village game logic

### ISSUE THREAD

**Issue by FICTURE7** [FICTURE7/CoCSharp#55]
These are the village game logic that should be implemented in the server. If you think I missed something or something does not fit let me know. :]

- [ ] Village
    - [ ] Traps/Buildings/Decorations 
        - [ ] Placement
        - [ ] Constructing
    - [ ] Traps/Buildings
        - [ ] Upgrading
    - [ ] Obstacles
        - [ ] Clearing
        - [ ] Generating (growth)
    - [ ] Troops
        - [ ] Training
        - [ ] Upgrading
    - [ ] Spells
        - [ ] Brewing
        - [ ] Upgrading
    - [ ] Workers
        - [ ] Workers management

**Comment by OrkhanAlikhanov** [comment]
What about boosting?

**Comment by FICTURE7** [comment]
Thanks for pointing that out! :smile:

**Comment by FastFrench** [comment]
Maybe you should also handle 
- broken/rearm for traps, 
- ammo level and refuel for defs like XBow or InfernoTower. 
- mode for skel traps, XBows, Inferno, Warden...

Up to you also if you want to handle the multiple layouts for the village, and war village.

**Comment by FICTURE7** [comment]
Yeah thanks for pointing that out again. :smiley: But for the multiple layouts for the village and war village we will see.

**Comment by ghost** [comment]
Clan castle. Empty or full.

**Comment by FICTURE7** [comment]
Yep, thanks!

**Comment by Hectoronian** [comment]
Maybe Defending mode for x-bows(Ground, Air/Ground) and infernos (Multiple, Single) which might go on buildings:

Maybe later sub-buildings (Defenders, Collectors, etc.)

PD: Never mind this, @FastFrench  pointed it out thanks!
